lib/operations/cursor_ops.js

```diff
@@ -20,39 +20,30 @@ function count(cursor, applySkipLimit, opts, callback) {
     if (typeof cursor.cursorSkip() === 'number') opts.skip = cursor.cursorSkip();
     if (typeof cursor.cursorLimit() === 'number') opts.limit = cursor.cursorLimit();
   }
+  if (cursor.s.cmd.hint) {
+    opts.hint = cursor.s.cmd.hint;
+  } else if (cursor.s.options.hint) {
+    opts.hint = cursor.s.options.hint;
+  }
+  if (
+    typeof opts.maxTimeMS !== 'number' &&
+    cursor.s.cmd &&
+    typeof cursor.s.cmd.maxTimeMS === 'number'
+  ) {
+    opts.maxTimeMS = cursor.s.cmd.maxTimeMS;
+  }
 
   // Command
   const delimiter = cursor.s.ns.indexOf('.');
+  opts.collectionName = cursor.s.ns.substr(delimiter + 1);
 
-  let command = {
-    count: cursor.s.ns.substr(delimiter + 1),
-    query: cursor.s.cmd.query
-  };
+  const command = createCountCommand(cursor, cursor.s.cmd.query, opts);
 
-  // Apply a readConcern if set
-  if (cursor.s.cmd.readConcern) {
-    command.readConcern = cursor.s.cmd.readConcern;
+  // Ensure we have the right read preference inheritance
+  if (opts.readPreference) {
```

Comment:
we have to be very careful here, this code depends on the fact that `createCountCommand` is mutating the passed in `opts`, and reading `readPreference` from it. I think we should avoid side effects at all costs, so two comments:
- Let's build a new `options` to be passed into the count command builder, plucking pieces of the input `opts` that we need
- Either use `resolveReadPreference` _before_ calling `createCoundCommand`, or have that method return a tuple of sorts (an `Array<command, options>` or `{ command, options }`)